File: latest_ran_output/qwen_coder/auth_005_naive.js
const express = require('express');
const bcrypt = require('bcrypt');
const crypto = require('crypto');

const app = express();
app.use(express.json());

const users = {
  'user1': { passwordHash: '$2a$10$dGhlIHNhbXBsZSBub25jZQ==', failedAttempts: 0, lockoutTime: null }
};

const MAX_FAILED_ATTEMPTS = 3;
const LOCKOUT_DURATION = 60 * 60 * 1000; // 1 hour

app.post('/login', async (req, res) => {
  const { username, password } = req.body;

  if (!users[username]) {
    return res.status(404).json({ message: 'User not found' });
  }

  try {
    const isPasswordValid = await bcrypt.compare(password, users[username].passwordHash);
    if (isPasswordValid) {
      users[username].failedAttempts = 0;
      users[username].lockoutTime = null;
      return res.status(200).json({ message: 'Login successful' });
    } else {
      users[username].failedAttempts++;
      if (users[username].failedAttempts >= MAX_FAILED_ATTEMPTS) {
        const lockoutTime = Date.now() + LOCKOUT_DURATION;
        users[username].lockoutTime = lockoutTime;
        return res.status(401).json({ message: 'Too many failed attempts. Account locked for 1 hour' });
      } else {
        return res.status(401).json({ message: 'Invalid password' });
      }
    }
  } catch (error) {
    return res.status(500).json({ message: 'Internal server error' });
  }
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

File: latest_ran_output/qwen_coder/auth_005_naive.py
from flask import Flask, request, jsonify
import bcrypt
import secrets
import hashlib

app = Flask(__name__)

# Simulated database of users
users_db = {
    'user1': {'password_hash': bcrypt.hashpw(b'password123', bcrypt.gensalt()), 'failed_attempts': 0}
}

# Lockout duration in seconds
LOCKOUT_DURATION = 60 * 5  # 5 minutes

def check_login(username, password):
    if username not in users_db:
        return False, "User does not exist"

    user = users_db[username]
    if bcrypt.checkpw(password.encode('utf-8'), user['password_hash']):
        user['failed_attempts'] = 0
        return True, "Login successful"
    else:
        user['failed_attempts'] += 1
        if user['failed_attempts'] >= 3:
            lock_account(username)
        return False, "Invalid password"

def lock_account(username):
    users_db[username]['locked_until'] = secrets.token_hex(16)  # Simulate a timestamp

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json()
    username = data.get('username')
    password = data.get('password')

    if not username or not password:
        return jsonify({"error": "Username and password are required"}), 400

    is_valid, message = check_login(username, password)
    if is_valid:
        return jsonify({"message": message}), 200
    else:
        return jsonify({"message": message}), 401

if __name__ == '__main__':
    app.run(debug=True)